### DISCUSSION TITLE
Stack/tree-based navigation use

### DISCUSSION THREAD

**Discussion by sillygoose** [pointfreeco/swift-composable-architecture#2445] [Q&A] [answered]
Updating a TCA app use all the wonderful new features in TCA 1.0 I find as I follow the latest Standup video series and have a couple of best-practices questions.

Standups uses stack-based navigation but after reading the stack-based  and tree-based documentation I am on the fence which way to go:

1. I have a tab-based app so if I went with stack-based navigation would I create separate paths for each tab?  It would seem so but I would to hear any ideas on this.

2.  I have updated a feature to include the unified navigation (aka `struct Destination`) for a sheet and an alert so it seems like an easy step to add .navigationDestination to the view and a new entry in Destination for the drill-down detail but I am stuck on the NavigationLink part (and I haven't found any sample code to help me out).

Like Standups I have an identified array of items that I would like to drill-down and show a detailed view but using the tree-based method.  The Standup code to do this is:
```
List {
  ForEach(viewStore.state) { standup in
    NavigationLink(
      state: AppFeature.Path.State.detail(StandupDetailFeature.State(standup: standup))
    ) {
      CardView(standup: standup)
    }
  }
}
```
If anyone can point out the changes I need to make the stack-based Standups code work in a tree-based world or can direct me to working code I would very much appreciate it.

**Comment by mbrandonw** [comment, marked as answer]
> 1. I have a tab-based app so if I went with stack-based navigation would I create separate paths for each tab?  It would seem so but I would to hear any ideas on this.

Yes, if you are using stack-based navigation, then each tab feature would maintain its own navigation path.

> 2. I have updated a feature to include the unified navigation (aka `struct Destination`) for a sheet and an alert so it seems like an easy step to add .navigationDestination to the view and a new entry in Destination for the drill-down detail but I am stuck on the NavigationLink part (and I haven't found any sample code to help me out).

If you are are targeting iOS <16, then you can use [`NavigationLinkStore`](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/navigationlinkstore) to drive navigation from a `Destination` enum. If you are targeting iOS >=16, then you can use our [`navigationDestination`](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/withviewstore/navigationdestination(store:state:action:destination:)) view modifier to drive navigation from a `Destination` enum.

We have a few examples of using these tools. For example, the ["Multiple Destinations" case study](https://github.com/pointfreeco/swift-composable-architecture/blob/f0bae57f361b3611b48d7db0b78393e5c78be1fc/Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-Multiple-Destinations.swift) and [PresentationTestCase.swift](https://github.com/pointfreeco/swift-composable-architecture/blob/f0bae57f361b3611b48d7db0b78393e5c78be1fc/Examples/Integration/Integration/PresentationTestCase.swift#L376-L383) integration test both show how to use `navigationDestination`. And [here's](https://github.com/pointfreeco/swift-composable-architecture/blob/f0bae57f361b3611b48d7db0b78393e5c78be1fc/Examples/Integration/Integration/PresentationTestCase.swift#L511-L513) how to use `NavigationLinkStore`.

**Reply by sillygoose** [reply]
Thanks for the reply and helpful suggestions, I develop for myself I am targeting iOS17 and the latest version of TCA learn the latest and greatest developments.  In the past I would work in the cold and dark but I came up with a new idea recently so dusted off some previous-generation TCA code and getting a head start.

After I posted this question I went back and reviewed some older episodes that I had skipped over and found what I needed in the Modern SwiftUI series, hopefully I can integrate the ideas there and complete the updating my older TCA code.

I see the final episode of the TCA 1.0 tour is live so I need to get busy completing the tour and applying it to my code.  I can't speak for anyone else but I find the Point-Free collections a fantastic resource for writing modern iOS/SwiftUI code.
